modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1295,15 +1330,14 @@ public void loadTableListenerToZoneReplica(
             Function<RaftCommandRunner, ReplicaListener> tablePartitionReplicaListenerFactory,
             RaftGroupListener tablePartitionRaftListener
     ) {
-        CompletableFuture<Replica> replicaFut = replicaMgr.replica(zonePartitionId);
-
-        assert replicaFut != null && replicaFut.isDone();
-
-        var zonePartitionReplicaListener = (ZonePartitionReplicaListener) replicaFut.join().listener();
+        Listeners listeners = listenersByZonePartitionId.get(zonePartitionId);
 
-        zonePartitionReplicaListener.addTableReplicaListener(tablePartitionId, tablePartitionReplicaListenerFactory);
+        listeners.replicaListenerFuture.thenAccept(zoneReplicaListener -> zoneReplicaListener.addTableReplicaListener(
```

Comment:
Please add a comment explaining why deadlock is not possible neither during node startup nor during common zone + table creation. I mean that both table creation and thus loading table processor to zone and zone creation occur within MS watch thread.

Reply:
I don't think I fully understand you, there can be no deadlock because we never wait for this future in any thread. I'll add some clarification comments